src/utils.js

```diff
@@ -8,16 +9,29 @@ function find(array, predicate) {
   }
 }
 
-function findHandler(handlers, method, url) {
+function findHandler(handlers, method, url, body) {
   return find(handlers[method.toLowerCase()], function(handler) {
     if (typeof handler[0] === 'string') {
-      return url === handler[0];
+      return url === handler[0] && isBodyMatching(body, handler[4]);
     } else if (handler[0] instanceof RegExp) {
-      return handler[0].test(url);
+      return handler[0].test(url) && isBodyMatching(body, handler[4]);
     }
   });
 }
 
+function isBodyMatching(body, requiredBody) {
+  var parsedBody;
+  var isJSON = true;
+  try {
+    parsedBody = JSON.parse(body);
```

Comment:
This code here is a bit ugly, but JSON.parse with a non json argument throws an error and the request body could also be a simple string for example..